corehq/apps/ota/views.py

```diff
@@ -32,6 +42,70 @@ def restore(request, domain, app_id=None):
     return get_restore_response(domain, couch_user, app_id, **get_restore_params(request))
 
 
+@json_error
+@login_or_digest_or_basic_or_apikey()
+def search(request, domain):
+    """
+    Accepts search criteria as GET params, e.g. "https://www.commcarehq.org/a/domain/phone/search/?a=b&c=d"
+    Returns results as a fixture with the same structure as a casedb instance.
+    """
+    criteria = dict(request.GET)
+    try:
+        case_type = criteria.pop('case_type')
+    except KeyError:
+        return HttpResponse('Search request must specify case type', status=400)
+
+    search_es = CaseSearchES()
+    search_es = search_es.domain(domain)
+    search_es = search_es.case_type(case_type)
+    config = CaseSearchConfig(domain=domain).config
+    fuzzies = config.get_fuzzy_properties_for_case_type(case_type)
+    for key, value in criteria.items():
+        search_es = search_es.case_property_query(key, value, fuzzy=(key in fuzzies))
+    results = search_es.values()
+    if should_use_sql_backend(domain):
+        fixtures = CaseDBFixture([CommCareCase.wrap(case) for case in results])
+    else:
+        cases = (CommCareCaseSQLSerializer(data=case).validated_data for case in results)
+        fixtures = CaseDBFixture([CommCareCaseSQL(**case) for case in cases])
```

Comment:
1. Lol. Thank you.
2. :+1: That sounds a lot more reasonable! (Django models have a `validated_data` property? I've never used that.)
3. Fair enough.